website/blog/2019-11-04-binding-react-navigation-5-to-ui-kitten.md

```diff
@@ -0,0 +1,429 @@
+
+Open `./src/navigators/auth.navigator.tsx` file and paste the following code:
+
+
+```js
+import React from 'react';
+import { createStackNavigator } from '@react-navigation/stack';
+import { SignInScreen, SignUpScreen, ResetPasswordScreen } from '@app-scenes/auth';
+import { AppRoute } from './app-routes';
+
+const Stack = createStackNavigator();
+
+export const AuthNavigator = (): React.ReactElement => (
+  <Stack.Navigator headerMode='none'>
+    <Stack.Screen name={AppRoute.SIGN_IN} component={SignInScreen}/>
+    <Stack.Screen name={AppRoute.SIGN_UP} component={SignUpScreen}/>
+    <Stack.Screen name={AppRoute.RESET_PASSWORD} component={ResetPasswordScreen}/>
+  </Stack.Navigator>
+);
+```
+
+In this example, we're using a `createStackNavigator` function to create simple stack navigation between Sign In, Sign Up and Reset Password screens. Under `Stack Navigator` we mean the default navigation behavior between screens: with slide-from-right animation on iOS, and slide-in-top on Android.
+
+In app.navigator.tsx file and replace the placeholder screen with Auth Navigator. This will make authentication screens to be the starter point of your app.
+
+```js
+import React from 'react';
+import { NavigationNativeContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import { AuthNavigator } from './auth.navigator';
+import { AppRoute } from './app-routes';
+
+const Stack = createStackNavigator();
+
+export const AppNavigator = (config): React.ReactElement => (
+  <NavigationNativeContainer>
+    <Stack.Navigator headerMode='none'>
+      <Stack.Screen name={AppRoute.AUTH} component={AuthNavigator}/>
+    </Stack.Navigator>
+  </NavigationNativeContainer>
+);
+```
+
+<img src="assets/binding-react-navigation-5-to-ui-kitten/stack-navigator.gif" height="420" />
+
+### Step 2. Top tabs
+
+Say, our app has both in-progress and finished tasks. So, you should separate them to avoid a mess. Here you can make it with two tabs on the home screen. To do this, we need to have three screens: two for tabs and one master screen for navigation management between tabs. Unlike the Stack Navigator component, the Top Tabs `Navigator` has a special prop for the component to control navigation between tabs - `tabBar`. We will use it to configure the tab bar with UI Kitten components.
+
+Open `./src/navigators/todo.navigator.tsx` file and paste the following code:
+
+```js
+import React from 'react';
+import { createStackNavigator } from '@react-navigation/stack';
+import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs';
+import { TodoScreen, TodoInProgressScreen, TodoDoneScreen } from '@app-scenes/todo';
+import { AppRoute } from './app-routes';
+
+const Stack = createStackNavigator();
+const TopTab = createMaterialTopTabNavigator();
+
+export const TodoNavigator = (): React.ReactElement => (
+  <TopTab.Navigator tabBar={TodoScreen}>
```

Comment:
`tabBar` accepts a render callback, not component:

```suggestion
  <TopTab.Navigator tabBar={props => <TodoScreen {...props} />}>
```

Reply:
That is exactly what it does, because `TodoScreen` [is a function](
https://github.com/artyorsh/react-navigation-ex-demo/blob/complete-exmaples/src/scenes/todo/todo.component.tsx#L26) :) Just inline syntax